src/Common/isLocalAddress.cpp

```diff
@@ -74,6 +79,33 @@ struct NetworkInterfaces
     {
         freeifaddrs(ifaddr);
     }
+
+    static std::shared_ptr<const NetworkInterfaces> instance()
```

Comment:
The code is too complex. Let's not update the list of network interfaces at all.
Let's make it a class and a single static variable instead of using multiple static variables and a mutex. 

The names `nf` and `mtx` are bad. Should be `network_interface` and `mutex`.